lib/trino-parquet/src/test/java/io/trino/parquet/reader/BenchmarkShortDecimalColumnReader.java

```diff
@@ -0,0 +1,210 @@
+        columnReader.setPageReader(new PageReader(UNCOMPRESSED, new LinkedList<>(dataPages), null, MAX_VALUES), null);
+        int rowsRead = 0;
+        while (rowsRead < MAX_VALUES) {
+            int remaining = MAX_VALUES - rowsRead;
+            columnReader.prepareNextRead(Math.min(READ_BATCH_SIZE, remaining));
+            rowsRead += columnReader.readPrimitive(field).getBlock().getPositionCount();
+        }
+    }
+
+    private DataPage createDataPage(ValuesWriter writer, int valuesCount)
+    {
+        Slice data;
+        try {
+            data = Slices.wrappedBuffer(writer.getBytes().toByteArray());
+            writer.reset();
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        return new DataPageV1(
+                data,
+                valuesCount,
+                byteArrayLength * valuesCount,
+                OptionalLong.empty(),
+                RLE,
+                RLE,
+                PLAIN);
+    }
+
+    private PrimitiveField getField()
+    {
+        int precision = maxPrecision(byteArrayLength);
+        PrimitiveType parquetType = Types.optional(FIXED_LEN_BYTE_ARRAY)
+                .length(byteArrayLength)
+                .as(LogicalTypeAnnotation.decimalType(0, precision))
+                .named("name");
+        return new PrimitiveField(
+                DecimalType.createDecimalType(precision),
+                true,
+                new ColumnDescriptor(new String[] {}, parquetType, 0, 0),
+                0);
+    }
+
+    private ValuesWriter createValuesWriter(int bufferSize)
+    {
+        return new FixedLenByteArrayPlainValuesWriter(byteArrayLength, bufferSize, bufferSize, HeapByteBufferAllocator.getInstance());
+    }
+
+    private void writeValue(ValuesWriter writer, long[] batch, int index)
+    {
+        Binary binary = Binary.fromConstantByteArray(longToBytes(batch[index], byteArrayLength));
+        writer.writeBytes(binary);
+    }
+
+    private long[] generateDataBatch(int size, int precision)
+    {
+        return unscaledRandomShortDecimalSupplier(byteArrayLength * Byte.SIZE, precision).apply(size);
+    }
+
+    private static int maxPrecision(int numBytes)
+    {
+        return toIntExact(
+                // convert double to long
+                Math.round(
+                        // number of base-10 digits
+                        Math.floor(Math.log10(
+                                Math.pow(2, 8 * numBytes - 1) - 1))));  // max value stored in numBytes
+    }
+
+    static {
+        // Pollute JVM profile
+        try {
+            for (int length = 1; length <= 8; length++) {
+                BenchmarkShortDecimalColumnReader benchmark = new BenchmarkShortDecimalColumnReader(length);
+                benchmark.setup();
+                benchmark.read();
+            }
+        }
+        catch (Throwable throwable) {
+            throw new RuntimeException(throwable);
+        }
+    }
```

Comment:
This is not a proper way to pollute the method profile. It's entirely possible the VM could ignore a single invocation, or that the profile collected during interpretation is independent of the one collected during C1. Use bulk warmup mode for this.

Reply:
> Should we be switching to to using bulk warm mode everywhere
...
> One drawback of the bulk warm mode way is that it increases the warm up time of the benchmark a lot 

For long running benchmarks, this will be prohibitively slow, as warmup is usually 30-70% of benchmark time